### ISSUE TITLE
Investigate problems of HIP builds

### ISSUE THREAD

**Issue by rasolca** [eth-cscs/DLA-Future#602]
https://github.com/eth-cscs/DLA-Future/pull/459#issuecomment-1179232091

**Comment by rasolca** [comment]
Update:
I rerun the test with the new image #592 (rocm5.1.3) and the results are:
a different error:
```
[ RUN      ] BacktransformationT2BTestGPU/0.CorrectnessLocal
/DLA-Future/test/unit/eigensolver/test_bt_band_to_tridiag.cpp:161: Failure
Failed
Error at index (1, 0): expected -0.571167 == -0.61407 (Relative diff: 0.0698675 > 2.86102e-06, Absolute diff: 0.0429035 > 2.86102e-06)

[  FAILED  ] BacktransformationT2BTestGPU/0.CorrectnessLocal, where TypeParam = float (11371 ms)
```
and again node killed by
`test_broadcast_panel`

**Comment by rasolca** [comment]
During last attempt to run the CI image test on ault (with AMD GPUs) for #642 RANK_[1,2,4] test passed, however some RANK_6 tests drains the node (including `test_cholesky` that was passing in the past).
Not sure about the full list of the failing and node draining tests as I stopped testing after I drained 3 nodes.

**Comment by rasolca** [comment]
Solved by #742.
